haralyzer/assets.py

```diff
@@ -60,12 +59,13 @@ def match_headers(self, entry, header_type, header, value, regex=True):
 
         :returns: a ``bool`` indicating whether a match was found
         """
-        if header_type not in entry:
+        entry = HarEntry(entry)
```

Comment:
I'm trying to reason about the need to cast `entry` to `HarEntry` here. When you initialize a `HarParser` or `HarPage`, each entry will already be a `HarEntry`. I'm guessing that you are trying to be as backwards compatible as possible (maybe somebody has code where they pass their own dict into `match_headers` instead of using the parser/page). I feel like we need to make this consistent and either assume that all of the entries are already `HarEntry` (the proper way to use the module), or make a decorator or something which we can apply to every method which will convert a dict if it was passed into a `HarEntry`. Thoughts?

Reply:
For the tests, the entries where being passed as a `dict()` so for those cases entry needs to be converted into a `HarEntry` into to use the attributes. 
I looked to doing a decorator but the issue is that they do not have access to any variables passed to the functions. If you know of a way that can get a shared way to do it I can implement it.

Reply:
Well, you CAN get access to the arguments inside of a decorator, but it would need to rely on the fact that the entry is the first arg. Here is a (potentially non-working) example.

```
import functools

def convert_to_entry(func):
    @functools.wraps(func)
    def inner(*args, **kwargs):
        # Convert the dict (first argument) to HarEntry
        if not isinstance(args[0], HarEntry):
            args[0] = HarEntry(args[0])
        return func(*args, **kwargs)
    return inner

@convert_to_entry
def thing_that_works_with_entries(entry,.....):
```
Not a huge deal but maybe worth a shot? I guess I just don't want you having to cast to `HarEntry` manually inside of each function if possible.